src/main/java/org/broadinstitute/hellbender/tools/sv/cluster/CNVLinkage.java

```diff
@@ -0,0 +1,103 @@
+package org.broadinstitute.hellbender.tools.sv.cluster;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.Genotype;
+import org.broadinstitute.hellbender.tools.sv.SVCallRecord;
+import org.broadinstitute.hellbender.tools.sv.SVCallRecordUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.variant.VariantContextGetters;
+
+import java.util.List;
+import java.util.Set;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.utils.GATKSVVCFConstants.COPY_NUMBER_FORMAT;
+
+/**
+ * Clustering engine class for defragmenting depth-based DEL/DUP calls, such as those produced by
+ * {@link org.broadinstitute.hellbender.tools.copynumber.GermlineCNVCaller GermlineCNVCaller}. Each variant is first padded by a fraction
+ * of its length and then merged with any other overlapping variants that meet the minimum sample overlap. Additionally,
+ * singleton variants (with only one carrier sample) will only be merged with variants of the same copy number.
+ */
+public class CNVLinkage extends CanonicalSVLinkage<SVCallRecord> {
```

Comment:
Ok I'm caving - I've made `SVClusterLinkage` an abstract superclass for `CanonicalSVLinkage` and `CNVLinkage` so that the unused pesr/mixed/depth params are no longer a part of `CNVLinkage`. I had to move a couple helper functions up into the parent but overall it is better this way (also found a couple of places like in the Factory where the depth parameters were still being passed in even though they aren't used for `CNVLinkage`).